### PR TITLE
Add folder lock API documentation

### PR DESCRIPTION
**Pull request by jcleblanc**
# Description

[DO NOT MERGE UNTIL FINAL PUSH DATE IS UPDATED]

Three new folder guide pages for the new folder lock API:

- GET a list of locks on a folder
- POST to create a new move / delete lock on a folder
- DELETE to remove an existing lock on a folder

This guidance will launch in conjunctions with 

- New API endpoints and objects
- cURL samples

Fixes # (issue)
Re `DDOC-278`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn lint` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream developer branch

### CODE REVIEW

File: content/guides/folders/single/create-lock.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
```
Comment (cbetta) [inline content/guides/folders/single/create-lock.md:1]: Might it be nice to put these under `folders/locks/create.md` etc?

Comment (cbetta) [reply content/guides/folders/single/create-lock.md:1]: Might be overthinking this so please take this with a grain of salt.

File: content/guides/folders/single/create-lock.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+rank: 8
+related_endpoints:
+  - post_folder_lock
+related_guides:
+  - folders/single/get-lock
+  - folders/single/delete-lock
+required_guides: []
+related_resources:
+  - folder-lock
```
Comment (cbetta) [inline content/guides/folders/single/create-lock.md:10]:
```suggestion
  - folder_lock
```

File: content/guides/folders/single/create-lock.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+`id` of the folder for which the lock should be applied. Optionally you may
+supply the specific `locked_operations` to be applied with the folder lock.
+
+<Samples id='post_folder_locks' />
+
+<Message type='warning'>
+  # Setting Locked Operations
+
+  If the `locked_operations` object is included with a folder lock request,
+  both `move` and `delete` need to be set to 'true'. Supplying only one lock
```
Comment (cbetta) [inline content/guides/folders/single/create-lock.md:26]:
```suggestion
  both `move` and `delete` need to be set to `true`. Supplying only one lock
```

File: content/guides/folders/single/delete-lock.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+rank: 10
+related_endpoints:
+  - delete_folder_locks_id
+related_guides:
+  - folders/single/create-lock
+  - folders/single/get-lock
+alias_paths: []
+related_resources:
+  - folder-lock
```
Comment (cbetta) [inline content/guides/folders/single/delete-lock.md:10]:
```suggestion
  - folder_lock
```

File: content/guides/folders/single/get-lock.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (cbetta) [inline content/guides/folders/single/get-lock.md:1]: Would this not be better called `get-locks`?

File: content/guides/folders/single/get-lock.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+rank: 9
+related_endpoints:
+  - get_folder_lock
```
Comment (cbetta) [inline content/guides/folders/single/get-lock.md:4]:
```suggestion
  - get_folder_locks
```

File: content/guides/folders/single/create-lock.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+---
+rank: 8
+related_endpoints:
+  - post_folder_lock
```
Comment (cbetta) [inline content/guides/folders/single/create-lock.md:4]:
```suggestion
  - post_folder_locks
```

File: content/guides/folders/single/get-lock.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+related_guides:
+  - folders/single/create-lock
+  - folders/single/delete-lock
+required_guides: []
+related_resources:
+  - folder-locks
```
Comment (cbetta) [inline content/guides/folders/single/get-lock.md:10]:
```suggestion
  - folder_locks
```

**Review by cbetta — CHANGES_REQUESTED** [review]

File: content/guides/folders/single/get-lock.md (hunk 5 above)
Comment (jcleblanc) [reply content/guides/folders/single/get-lock.md:1]: Good point, modified

File: content/guides/folders/single/create-lock.md (hunk 1 above)
Comment (jcleblanc) [reply content/guides/folders/single/create-lock.md:1]: I'll keep it as is for now as I think the list length is pretty good right now. If we add another series of APIs for folder operations let's revisit this and compartmentalize the content a bit more.

**Comment by jcleblanc** [comment]
@cbetta changes are up, thanks for the review.

**Review by cbetta — APPROVED** [review]

**Pull request merged**